mat_nn_kokkos.hpp

```diff
@@ -66,17 +78,29 @@ double su3_mat_nn(h_site_view &a, h_su3_matrix_view &b, h_site_view &c,
         printf("Device number set to %d\n", use_device);
     }
 
+    start.reset();
+    auto tprofiling = Clock::now();
+
     d_site_view d_a("d_a", total_sites);
     d_site_view d_c("d_c", total_sites);
     d_su3_matrix_view d_b("d_b", 4);
 
     Kokkos::deep_copy(d_a, a);
     Kokkos::deep_copy(d_b, b);
 
-    double ttotal = k_mat_nn(iterations, d_a, d_b, d_c, total_sites,
-                             blocksPerGrid, threadsPerBlock);
+    profile->h2d_time = (std::chrono::duration_cast<std::chrono::microseconds>(Clock::now()-tprofiling).count())/1.0e6;
+    tprofiling = Clock::now();
```

Comment:
Instead of setting tprofiling here I think we need to pass a pointer to it into k_mat_nn and then set it after the warmups are complete. As it stands this will include the warmup iterations in the timer.